### PR TITLE
Do not remove locked data in inventory prepare

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

Removing locked data in inventory preparation causes values not to be stored in locked fields table.
This PR removes that, but still prevents locked relations to be created.

### CODE REVIEW

**Comment by cedric-anne** [comment]
In this specific context, I guess that used locked values should be raw values.

Maybe the logic that replaces dropdown ids by their corresponding name should be moved in the twig template, so that the `getLockedValues` method would return values that can be used in any context.

**Review by stonebuzz — APPROVED** [review]

File: src/Lockedfield.php [hunk 1]
```diff
@@ -178,16 +178,7 @@ public function getLocks($itemtype, $items_id, bool $fields_only = true)
...
             } else {
-                $value = $row['value'];
-                //if it refers to a ForeignKey get dropdown name
-                if (
-                    $row['value'] !== null
-                    && isForeignKeyField($row['field'])
-                    && is_a($itemtype = getItemtypeForForeignKeyField($row['field']), CommonDropdown::class, true)
-                ) {
-                    $value = Dropdown::getDropdownName(getTableForItemType($itemtype), $row['value'], 0, true, false, $row['value']);
-                }
```
Comment (cedric-anne) [inline src/Lockedfield.php:189]: This logic should be put back somewhere (in the twig template maybe), in order to hace dropdown names displayed in lock icon tooltip.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/Lockedfield.php (hunk 1 above)
Comment (trasher) [reply src/Lockedfield.php:189]: In my tests, all values were displayed corrdctly. We do not want to display dropdowns names here, but inventory value.

Comment (trasher) [reply src/Lockedfield.php:189]: I just add such a case in test suite.

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**